app/static/src/app/components/help/DraggableDialog.vue

```diff
@@ -0,0 +1,145 @@
+<template>
+  <div ref="draggable" class="draggable-dialog p-2" :style="dialogStyle">
+    <div class="dragtarget" @mousedown="handleDragStart" @touchstart="handleDragStart">
+      <h3 class="move prevent-select">
+         <vue-feather type="move" class="grey"></vue-feather>
+          {{title}}
+        <vue-feather type="x" class="clickable grey float-end" @click="close"></vue-feather>
+      </h3>
+    </div>
+    <div class="overflow-auto draggable-content">
+      <slot></slot>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import {
+    computed, defineComponent, onMounted, ref
+} from "vue";
+import VueFeather from "vue-feather";
+
+interface Point {
+  x: number,
+  y: number
+}
+
+export default defineComponent({
+    name: "DraggableDialog",
+    props: {
+        title: String
+    },
+    components: {
+        VueFeather
+    },
+    setup(props, { emit }) {
+        const draggable = ref<null | HTMLElement>(null); // Picks up the element with 'draggable' ref in the template
+
+        const position = ref<Point | null>(null);
+        const moveClientStart = ref<Point | null>(null);
+        const movePositionStart = ref<Point | null>(null);
+
+        const getTouchEvent = (event: Event) => {
+            return (event as TouchEvent).touches?.length > 0 ? (event as TouchEvent).touches[0] : event;
+        };
+
+        const draggableRect = () => (draggable.value ? draggable.value.getBoundingClientRect() : {
+            left: 0, top: 0, width: 0, height: 0
+        });
+
+        const containInWindow = (newPosition: Point) => {
+            // Don't allow top of draggable to move outside the bounds of the window
+            const rect = draggableRect();
+            const viewportWidth = window.innerWidth;
+            const viewportHeight = window.innerHeight;
+
+            const containedX = Math.min(Math.max(newPosition.x, 0), viewportWidth - rect.width);
+            const containedY = Math.min(Math.max(newPosition.y, 0), viewportHeight - rect.height);
+
+            return { x: containedX, y: containedY };
+        };
+
+        const handleDragMove = (event: Event) => {
+            if (movePositionStart.value && moveClientStart.value) {
+                const { clientX, clientY } = getTouchEvent(event) as MouseEvent | Touch;
+                position.value = containInWindow({
+                    x: movePositionStart.value.x + clientX - moveClientStart.value.x,
+                    y: movePositionStart.value.y + clientY - moveClientStart.value.y
+                });
+            }
+            event.preventDefault();
```

Comment:
what does this do? I've found [these docs](https://vuejs.org/guide/essentials/event-handling.html#event-modifiers) but am still a bit in the dark

Reply:
It should just stop the event from resulting in any default behaviour on top of what's coded in our handler. So in this case I think it should stop text in the main window from being selected etc.